lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -68,8 +73,13 @@ def execute(self) -> int:
                 for path in tg_dir:
                     host = path.name
                     tool_info[group][host] = {}
+                    label = path / "__label__"
+                    tool_info[group][host]["label"] = (
+                        None if not label.exists() else label.read_text().rstrip("\n")
+                    )
+                    tool_info[group][host]["tools"] = {}
                     for tool in sorted(self.tools(path)):
-                        tool_info[group][host][tool] = (
+                        tool_info[group][host]["tools"][tool] = (
                             sorted((path / tool).read_text().rstrip("\n").split("\n"))
                             if opts
                             else ""
```

Comment:
I realize that you didn't write this, but it's kind of ugly.  Perhaps,
```python
                    tool_info[group][host]["tools"] = {
                        tool: sorted((path / tool).read_text().rstrip("\n").split("\n"))
	                if opts
                        else ""
                        for tool in sorted(self.tools(path))
                    }
```
but, I don't really like that, either.  How about this?
```python
                    tool_info[group][host]["tools"] = {}
                    for tool in sorted(self.tools(path)):
                        v = sorted((path / tool).read_text().rstrip("\n").split("\n"))
                        tool_info[group][host]["tools"][tool] = v if opts else ""
```

A similar comment applies to lines 120-127.

Reply:
I think I *did* write this, but I agree it's ugly :smile:

Reply:
I'm going with the last one, although it's doing a bit of unnecessary work in the "normal" case (without -o). Is that OK with you?  If not, I could use something like this:

```
                    tool_info[group][host]["tools"] = {}
                    for tool in sorted(self.tools(path)):
                        v = ""
                        if opts:
                             v = sorted((path / tool).read_text().rstrip("\n").split("\n"))
                        tool_info[group][host]["tools"][tool] = v
```

and split the long `v` calculation into two assignments.

Reply:
I like the one you picked.  I don't think it's doing unnecessary work (or, at least, any work which could be avoided) -- I think the others do basically the same.